src/battle_arcade_game_board.c

```diff
@@ -0,0 +1,1442 @@
+    }
+    
+    work->unk13_1++;
+    if (work->unk13_1 >= 4) {
+        work->unk13_1 = 0;
+        work->unk13_0 ^= 1;
+    }
+    
+    sub_0202457C(work->unk3E8.spriteList);
+    
+    return FALSE;
+}
+
+BOOL ov84_0223DFF0(OVY_MANAGER *man, int *state) {
+    int i;
+    GAME_BOARD_WORK *work = OverlayManager_GetData(man);
+    
+    *work->returnWork = work->cursorPos;
+    
+    GF_DestroyVramTransferManager();
+    
+    BattleArcadeGameBoard_Delete(work);
+    
+    OverlayManager_FreeData(man);
+    
+    Main_SetVBlankIntrCB(NULL, NULL);
+    DestroyHeap(HEAP_ID_GAME_BOARD);
+    UnloadOverlayByID(FS_OVERLAY_ID(OVY_80));
+
+    return TRUE;
+}
+
+static BOOL BattleArcadeGameBoard_Init(GAME_BOARD_WORK *work) {
+    switch (work->substate) {
+    case 0:
+        if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+            sub_02037BEC();
+            sub_02037AC0(199);
+            work->substate++;
+        } else {
+            work->substate++;
+        }
+        break;
+    case 1:
+        if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+            if (sub_02037B38(199) == TRUE) {
+                sub_02037BEC();
+                work->substate++;
+            }
+        } else {
+            work->substate++;
+        }
+        break;
+    case 2:
+        ov84_0223EE74(work);
+        work->substate++;
+        break;
+    case 3:
+        ov84_0223EFE0(work);
+        work->substate++;
+        break;
+    case 4:
+        ov84_0223F094(work);
+        work->substate++;
+        break;
+    case 5:
+        if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+            sub_02037BEC();
+            sub_02037AC0(201);
+            work->substate++;
+        } else {
+            work->substate++;
+        }
+        break;
+    case 6:
+        if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+            if (sub_02037B38(201) == TRUE) {
+                sub_02037BEC();
+                work->substate++;
+            }
+        } else {
+            work->substate++;
+        }
+        break;
+    case 7:
+        if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+            if (ov84_0223ECBC(work, 30, 0) == TRUE) {
+                work->substate++;
+            }
+        } else {
+            BeginNormalPaletteFade(0, 1, 1, 0, 6, 3, HEAP_ID_GAME_BOARD);
+            work->substate++;
+        }
+        break;
+    case 8:
+        if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+            if (work->unkF >= 2) {
+                work->unkF = 0;
+                BeginNormalPaletteFade(0, 1, 1, 0, 6, 3, HEAP_ID_GAME_BOARD);
+                work->substate++;
+            }
+        } else {
+            work->substate++;
+        }        
+        break;
+    case 9:
+        if (IsPaletteFadeFinished() == TRUE) {
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+static BOOL BattleArcadeGameBoard_MoveCursor(GAME_BOARD_WORK *work) {
```

Comment:
optional: with static methods, you probably don't need the `BattleArcadeGameBoard` prefix